### PR TITLE
Update unAssemble and Destructor for DEN

### PR DESCRIPTION
**Pull request by SoumiKDaS1701**

### CODE REVIEW

File: src/SparseMatrix_DEN.hpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        /*
+        Code for implementing getFormat for DEN
+        */
+        //
+        if (this->_state == assembled):
+          _unAssemble();
+        //
+        SparseMatrix<fp_type> B;
+        SparseMatrix<fp-type>* ptr_B = B;
+        // newMat -> _nrows, _ncols, _nnz, _buildCoeff = mat->
```
Comment (meichenl95) [inline src/SparseMatrix_DEN.hpp:88]: Delete unnecessary comments.

File: src/SparseMatrix_DEN.hpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+/**************/
+/* unAssemble */
+/**************/
+
+    template <class fp_type>
+    void SparseMatrix_DEN<fp_type>::unAssemble()
+    {
+        assert(this->_state == assembled);
+
```
Comment (meichenl95) [inline src/SparseMatrix_DEN.hpp:136]: Add some comments to make it clear.

**Review by meichenl95 — COMMENTED** [review]
Review from Meichen done.

**Review by meichenl95 — APPROVED** [review]

File: src/SparseMatrix_DEN.hpp (hunk 1 above)
Comment (SoumiKDaS1701) [reply src/SparseMatrix_DEN.hpp:88]: Comments deleted

File: src/SparseMatrix_DEN.hpp (hunk 2 above)
Comment (SoumiKDaS1701) [reply src/SparseMatrix_DEN.hpp:136]: Added some comments

**Pull request merged**